### PR TITLE
Use libunwind instead of libexecinfo

### PR DESCRIPTION
**Pull request by gquintard**
offered as an alternative to #3034

### CODE REVIEW

**Comment by nigoroll** [comment]
I have not looked at the patch, but in general I would hope that adding libunwind would help.

From a private email as of 2019-11-14:

[...] But what I would hope to get is:

* stack frame addresses for the backtrace

* portable information about stack boundaries to avoid stack overflows (e.g. before calling pcre with jit, we know we'll need ~33k at least, so we could bail out before even calling it if we know stack won't suffice or fall back to some less efficient alternative allocation)

* _maybe_ a nicer/portable longjmp replacement to handle stack overflows from VCL and _maybe_ also stuff we can't handle otherwise like SIGFPE

**Comment by gquintard** [comment]
note: force-pushed to edit the travis file

**Comment by bsdphk** [comment]
Do we want a hard switch or a period with the ability to pick which one ?

**Comment by bsdphk** [comment]
We will move on this after the 6.3 release.

From the bugwash discussion it was not very obvious what the portability situation was, only that one probably exists.

Initially we add a configure option to use libunwind, default to status quo, then when we have portability data, we decide what to do next.

File: .travis.yml [hunk 1]
```diff
@@ -18,9 +18,10 @@ jobs:
             - nghttp2
             - python3-docutils
             - python3-sphinx
+            - libunwind-dev
       before_script:
         - ./autogen.sh
-        - ./configure
+        - ./configure --with-unwind
```
Comment (fgsch) [inline .travis.yml]: Can you also add it below?

**Comment by gquintard** [comment]
thanks @fgsch, force-pushed

**Comment by fgsch** [comment]
For macos things are a bit different - the library is already there but no .pc file or headers.
So perhaps for now we want to remove it from there.

**Comment by gquintard** [comment]
gotcha, removing it then

**Comment by gquintard** [comment]
@bsdphk : forgot to mention I force-pushed an update. Default is to keep `execinfo`, using `--with-unwind` replaced it with the new kid

**Comment by gquintard** [comment]
OK'd by Poul-Henning

**Pull request merged**